### ISSUE TITLE
Insets with new line - the texts is not converted

### ISSUE THREAD

**Issue by MayaBloodsAI** [VisualSystemsCorp/vsc_quill_delta_to_html#9]
Hi (:,

I try to convert a json that contained a new line, the html I get contained only the new line with out the text:

My code:

`
    final deltaJson = delta.toJson();
    
    final converter = QuillDeltaToHtmlConverter(
      List.castFrom(deltaJson),
      ConverterOptions.forEmail(),
    );

    final text = converter.convert();
`

the json is:

`
{
          'insert': 'Dear Elise\n',
          'attributes': {'header': 1, 'align': 'center'}
}
`
** new line after "Elise"

And the html I get is:

`<h1 class="ql-align-center"><br/><br/></h1>`

when I add another inserts with out a new line I can see the test.. so maybe the problem is with the new line.

Thanks,

**Comment by CatHood0** [comment]
I test delta several times, and a rule that always make me cry, is the rules of always must be a `{"insert":"\n"},` if you want to do a new line after any text. Well, this in my case. 

That Json is created by you manually or comes from Flutter Quill (or another editor with delta)?

**Comment by MayaBloodsAI** [comment]
Hi ..
Yes, I get the text from flutter_quill editor, I just try to manually change the insert, and find out that it does work with out the attributes:
`
[
        {
          'insert': "Dear Elise\n",
        },
]
`

will give a correct html

`<p>Dear Elise</p>
`

But when attribute added, any attribute it doesn't include the text in the html..

**Comment by CatHood0** [comment]
Flutter quill works with strange delta and documents (His examples lie, not works like that). When you need block attributes (like: align, image, custom, header, etc), you must to divide the insert in two parts:

#### The first one: must to be the text that you want to have the block attributes (this insert must contain inline attributes). It should be something as: 

```dart
  {"insert":"Dear Elise"} //add inline attributes only here
```

#### The second one: must to be a insert (it CANNOT be empty), with a new line insertion, and his attributes must to be the block attributes that you want. It seems like: 

 ```dart
   {"insert":"\n","attributes":{"header":"2","align":"center"}} 
//don't add any inline attribute here (it doesn't apply)
```

#### In conclusion your delta should something like:
```dart
{"insert":"Dear Elise"},{"insert":"\n","attributes":{"header":"2","align":"center"}}
```
That's simple, but i don't get it until  when test several times and it works finally.

#### And, i noticed that you are having a error when you want add a new line. Flutter quill not add any new line correctly when you add a \n and a text in the same insertion. It's better make something like this delta Json:

 ```dart
  {"insert":"Dear Elise"},{"insert":"\n"} 
//don't add any inline attribute or block attributes here (it doesn't apply) 
```

**Comment by dsyrstad** [comment]
@MayaBloodsAI I'll echo a lot of what @CatHood0 said. If you write your document in Flutter Quill with all of the attributes you specified, the json looks like:

```json
[
  {
    "insert": "Dear Elise"
  },
  {
    "insert": "\n",
    "attributes": {
      "header": 1,
      "align": "center"
    }
  }
]
```

Which gives you:
```html5
<h1 style="text-align:center">Dear Elise</h1>
```

The block attributes must appear separate from the text with a `\n` insert. See the Quill JS documentation that defines the delta format: https://quilljs.com/docs/delta/#line-formatting. Your json needs to conform to the Quill Delta format in order to convert correctly.

**Comment by MayaBloodsAI** [comment]
You correct!! Thanks a lot @CatHood0  & @dsyrstad!!
